src/_P039_Thermocouple.ino

```diff
@@ -350,109 +629,97 @@ float readMax31855(struct EventStruct *event)
 
 float readMax31856(struct EventStruct *event)
 {
-  digitalWrite(Plugin_039_Get_SPI_CS_Pin(event), LOW);
-
-
-  # define P039_RAWVALUE 0
-  # define P039_CR0      1
-  # define P039_CR1      2
-  # define P039_MASK     3
-  # define P039_CJHF     4
-  # define P039_CJLF     5
-  # define P039_LTHFTH   6
-  # define P039_LTHFTL   7
-  # define P039_LTLFTH   8
-  # define P039_LTLFTL   9
-  # define P039_CJTO    10
-  # define P039_CJTH    11
-  # define P039_CJTL    12
-  # define P039_LTCBH   13
-  # define P039_LTCBM   14
-  # define P039_LTCBL   15
-  # define P039_SR      16
+  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
+
 
   uint32_t registers[17] = { 0 };
 
-  for (int i = 0; i < 17; ++i) {
-    registers[i] = SPI.transfer(0x0);
+  for (int i = 0u; i < 17; ++i) {
```

Comment:
That 17 seems like a magic number, as you've been using `#define`s liberally, this one probably deserves its own `#define P039_REGISTER_COUNT 17`

Reply:
Hm, usually 42 is said to be the "magic" number and the answer to all questions...

Let me check why i missed the #define opportunity here...and improve.

Reply:
> 42 is said to be the "magic" number and the answer to all questions...

Acknowledged, but any numeric value in code is deemed as a magic number by static code analysis tools, and a possible cause for bugs as it may not get updated if the range changes.

NB: TD-er pointed me once at another perfect number when I updated P073 and accidentally used P037 in a commit message:

> How could you mess up such a [perfect numer](https://bigbangtheory.fandom.com/wiki/73) ;)

😄

Reply:
What static code analysis tool can be used with ESpEasy development. Preferably it should be available as open source ?

Reply:
This occurence of a magic number has been closed with latest commit,